chainstate/src/detail/tests/test_framework.rs

```diff
@@ -65,11 +60,18 @@ impl BlockTestFramework {
         Self::with_chainstate(chainstate)
     }
 
-    pub fn random_block(&self, parent_block: &Block, params: Option<&[TestBlockParams]>) -> Block {
-        let (mut inputs, outputs): (Vec<TxInput>, Vec<TxOutput>) =
-            parent_block.transactions().iter().flat_map(create_new_outputs).unzip();
+    pub(in crate::detail::tests) fn random_block(
```

Comment:
Is the `pub(in crate::detail::tests)` part really needed? Isn't `BlockTestFramework` itself only visible in these tests?